### PR TITLE
Improve typing experience with targetted decorators

### LINKED ISSUE #7475: Bump @orge-tools/* versions
Motivation:
- The new version include fixes and improvements to the typings around decorators and targeted decorators.
- Will add a flag for cycle detection (instead of just call stack exhaustion)

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

### CODE REVIEW

**Review by jansav — DISMISSED** [review]

**Comment by Nokel81** [comment]
NOTE: this is blocked on a bug in ogre-tools relating to the typing of the function for non-targetted decoration.

**Comment by Iku-turso** [comment]
I see failing unit test?

>NOTE: this is blocked on a bug in ogre-tools relating to the typing of the function for non-targetted decoration.

Is this still relevant?

File: packages/core/src/features/telemetry/renderer/telemetry-decorator-for-show-details.injectable.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          },
+        });
+
+        return showDetails(...args);
+      };
+    },
+  }),
+  decorable: false,
+  injectionToken: instantiationDecoratorToken,
+});
```
Comment (Iku-turso) [inline packages/core/src/features/telemetry/renderer/telemetry-decorator-for-show-details.injectable.ts]: Unit tests missing, at least from commit.

File: packages/core/src/features/telemetry/renderer/telemetry-decorator-for-show-details.injectable.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+      return (...args) => {
+        emitTelemetry({
+          action: showDetailsInjectable.id,
+          params: {
+            kind: (() => {
+              try {
+                return parseKubeApi(args[0] || "").resource;
+              } catch {
+                return "";
```
Comment (Iku-turso) [inline packages/core/src/features/telemetry/renderer/telemetry-decorator-for-show-details.injectable.ts]: Nit: this is change in behaviour, previously errors where made known as part of "non-basic" telemetry.

Not accounting for if that's good or bad, commit name should indicate changes in behaviour.

File: packages/core/src/main/getDiForUnitTesting.ts [hunk 3]
```diff
@@ -25,18 +25,19 @@ import waitUntilBundledExtensionsAreLoadedInjectable from "./start-main-applicat
...
+import { setLegacyGlobalDiForExtensionApi } from "../extensions/as-legacy-globals-for-extension-api/legacy-global-di-for-extension-api";
 import { registerMobX } from "@ogre-tools/injectable-extension-for-mobx";
 import { registerFeature } from "@k8slens/feature-core";
 import { messagingFeature, testUtils as messagingTestUtils } from "@k8slens/messaging";
 
 export function getDiForUnitTesting() {
-  const di = createContainer("main");
+  const environment = "main";
+  const di = createContainer(environment, {
+    detectCycles: false,
```
Comment (Iku-turso) [inline packages/core/src/main/getDiForUnitTesting.ts:35]: Yikes! Let's fix this and fast.

File: packages/core/src/features/telemetry/renderer/telemetry-decorator-for-show-details.injectable.ts (hunk 1 above)
Comment (Iku-turso) [reply packages/core/src/features/telemetry/renderer/telemetry-decorator-for-show-details.injectable.ts]: Found 'em. Good policy for thought: include related tests in commit :)

File: packages/core/src/features/telemetry/renderer/telemetry-decorator-for-show-details.injectable.ts (hunk 2 above)
Comment (Iku-turso) [reply packages/core/src/features/telemetry/renderer/telemetry-decorator-for-show-details.injectable.ts]: This was restored in later commit, I suppose.

Comment (Nokel81) [reply packages/core/src/features/telemetry/renderer/telemetry-decorator-for-show-details.injectable.ts]: Moved and squashed some commits to make this all part of one commit

File: packages/core/src/features/telemetry/renderer/telemetry-decorator-for-show-details.injectable.ts (hunk 1 above)
Comment (Nokel81) [reply packages/core/src/features/telemetry/renderer/telemetry-decorator-for-show-details.injectable.ts]: Moved and squashed some commits to make this all part of one commit

**Review by gabriel-mirantis — DISMISSED** [review]

**Review by gabriel-mirantis — APPROVED** [review]

**Pull request merged**